### PR TITLE
Fix pep8 and cyclomatic complexity in _mkl

### PR DESCRIPTION
**Pull request by jakelishman**
This simplifies some of the long if/else blocks in the MKL module, making it simpler to parse.  No logic should actually have changed.

Everything is at most 10 cyclomatic complexity now, but I don't think that corresponds exactly to CodeClimate's metrics, so let's see.

**Changelog**: simplify logic in `_mkl` module.

### CODE REVIEW

File: qutip/_mkl/spsolve.py [hunk 1]
```diff
@@ -127,95 +136,127 @@ def __init__(self, np_pt=None, dim=None, is_complex=None,
...
+    11: 'Real non-symmetric',
+    13: 'Complex non-symmetric',
+}
+
+
+def _mkl_matrix_type(dtype, solver_args):
+    mtype = (
+        (0b0100 if dtype == np.complex128 else 0b0010)
+        | (0b0000 if solver_args['hermitian'] else 0b1001)
+    )
```
Comment (Ericgig) [inline qutip/_mkl/spsolve.py:250]: Why use bytes and not int?
```
    mtype = (
        (4 if dtype == np.complex128 else 2)
        + (0 if solver_args['hermitian'] else 9)
    )
```

File: qutip/tests/test_mkl.py [hunk 2]
```diff
@@ -31,205 +31,120 @@
...
+            [0, 0, 0],
+        ], dtype=dtype)
+        sX = mkl_spsolve(sM, N, verbose=True)
+        X = scipy.linalg.solve(M, N)
+        np.testing.assert_allclose(X, sX)
+
+    def test_rhs_shape_is_maintained(self):
+        """
+        MKL spsolve : Solution shape same as input RHS vec
+        """
```
Comment (Ericgig) [inline qutip/tests/test_mkl.py:88]: Why keep the doc-string only here?

File: qutip/_mkl/spsolve.py (hunk 1 above)
Comment (jakelishman) [reply qutip/_mkl/spsolve.py:250]: I had (erroneously, it turns out) assumed that this was interpreted as a bit field.  It's actually just specific numbers.  I'll change it to be more legible.

File: qutip/tests/test_mkl.py (hunk 2 above)
Comment (jakelishman) [reply qutip/tests/test_mkl.py:88]: A mistake - in my editor I have docstrings folded unless I open them, and I just overlooked it.  The docstrings don't add any information that we don't get from the names of the tests, which is why I removed them.

**Review by Ericgig — APPROVED** [review]

**Pull request merged**